odxtools/database.py

```diff
@@ -39,20 +37,23 @@ def __init__(self,
         self._comparam_subsets = NamedItemList[ComparamSubset]()
         self._comparam_specs = NamedItemList[ComparamSpec]()
 
-    def add_pdx_file(self, pdx_file_name: str) -> None:
-        pdx_zip = ZipFile(pdx_file_name)
-
-        for zip_member in pdx_zip.namelist():
+    def add_pdx_file(self, pdx_file: Union[str, PathLike[str], IO[bytes], ZipFile]) -> None:
```

Comment:
`PathLike` is not a generic. i.e.
```suggestion
    def add_pdx_file(self, pdx_file: Union[str, PathLike, IO[bytes], ZipFile]) -> None:
```

Reply:
(might be my mistake. if so: sorry)

Reply:
sorry, that was my mistake

Reply:
actually, we were both correct and both wrong: now the type checker (mypy) fails. this is because the python and the type checking signatures seem to be incompatible: https://stackoverflow.com/questions/55076778/why-isnt-this-function-type-annotated-correctly-error-missing-type-parameters

I guess, we should go with the third solution of the top answer of that question and specify that type as string, i.e.,
```python
    def add_pdx_file(self, pdx_file: Union[str, "PathLike[Any]", IO[bytes], ZipFile]) -> None:
```

(also for the other function signatures)